src/server/engine_shard_set.cc

```diff
@@ -139,7 +143,11 @@ void EngineShardSet::PreShutdown() {
 }
 
 void EngineShardSet::Shutdown() {
+  namespaces->Clear();
   RunBlockingInParallel([](EngineShard*) { EngineShard::DestroyThreadLocal(); });
+
+  delete namespaces;
+  namespaces = nullptr;
```

Comment:
You already call Clear in destructor 🤔 Also maybe just use an optional instead of a pointer to avoid managing it's value (` = nullptr`) and avoid double indirection (because namespaces just contains a pointer to the default namespace)

Reply:
The only oddness I see here is that `Clear()` is called before `DestroyThreadLocal()`, which perhaps warrants a comment to explain why. Otherwise this area looks good to me...
@dranikpg keep in mind that `std::optional` is not a POD, which kind of beats the purpose of this PR